### PR TITLE
[KVM_ON_S390] support guest installation on s390x lpar via virt-install

### PR DESCRIPTION
**Pull request by guoxuguang**
Try to use svirt BACKEDN to trigger automation Guests Installation test run on s390x LPAR

found that in svirt console we don't have a serial line and can't rely on assert_script_run check, so try to create new lpar_cmd and lpar_cmd_with_retry funs to trigger cmd, also use upload_asset to upload log to compare cmd output.

- Needles: no need
- Verification run: 
S390 GUEST INSTALLATION  SLES15SP1
http://10.67.17.251/tests/21
X86 GUEST INSTALLATION  SLES15SP1 
http://10.67.17.251/tests/17
X86 HOST UPGRADE SLES15->SLES15SP1
http://10.67.17.251/tests/32

### CODE REVIEW

**Comment by guoxuguang** [comment]
@alice-suse @waynechen55 @Julie-CAO please help me review this PR

File: tests/virt_autotest/virt_utils.pm [hunk 1]
```diff
@@ -48,14 +48,26 @@ sub repl_repo_in_sourcefile {
     my $verorig = "source.http.sles-" . get_version_for_daily_build_guest . "-64";
     my $veritem = check_var('ARCH', 'x86_64') ? $verorig : get_required_var('ARCH') . ".$verorig";
     if (get_var("REPO_0")) {
-        my $location = &virt_autotest_base::execute_script_run("", "perl /usr/share/qa/tools/location_detect_impl.pl", 60);
-        $location =~ s/[\r\n]+$//;
+	my $location = "";
+	if (check_var('ARCH', 's390x')) {
+            my $location = type_string("perl /usr/share/qa/tools/location_detect_impl.pl\n");
```
Comment (alice-suse) [inline tests/virt_autotest/virt_utils.pm]: For s390, type_string will not return real location, then the sourcefile variable will be wrong?

File: tests/virt_autotest/virt_utils.pm [hunk 2]
```diff
@@ -198,4 +217,59 @@ sub clean_up_red_disks {
...
 
+sub lpar_cmd {
+    my ($cmd, $mesg, $args) = @_;
+    die 'Command not provided' unless $cmd;
+    my $mesg = shift;
+
+    $args->{ignore_return_code} ||= 0;
+    my $ret = console('svirt')->run_cmd($cmd);
+    if ($ret == 0){
+        diag "Command $mesg on S390X LPAR returned: SUCESS";
```
Comment (alice-suse) [inline tests/virt_autotest/virt_utils.pm]: If you want diagnostic msg from openqa webui, then use record_info api can help.

File: tests/virt_autotest/virt_utils.pm [hunk 3]
```diff
@@ -198,4 +217,59 @@ sub clean_up_red_disks {
...
+    }
+    unless ($args->{ignore_return_code} || !$ret) {
+        diag "Command $mesg on S390X LPAR returned: FAIL";
+        die 'Find new failure, please check manually';
+    }
+}
+
+sub lpar_cmd_with_retry {
```
Comment (alice-suse) [inline tests/virt_autotest/virt_utils.pm]: This is never used in your code?

File: tests/virt_autotest/virt_utils.pm (hunk 2 above)
Comment (guoxuguang) [reply tests/virt_autotest/virt_utils.pm]: yes, try to use record_info api  as your mentioned, It does work to me.

Will add this record_info api. 
Test Run with record_info api 
http://10.67.17.251/tests/47

File: tests/virt_autotest/virt_utils.pm (hunk 3 above)
Comment (guoxuguang) [reply tests/virt_autotest/virt_utils.pm]: lpar_cmd_with_retry api work for install_package.pm and update_package.pm

**Pull request merged**